packages/paste-website/src/pages/primitives/combobox-primitive/index.mdx

```diff
@@ -0,0 +1,293 @@
+
+##### `defaultIsOpen?: boolean`
+
+This is the value to set the `isOpen` to anytime a combobox is reset, when the the selection is
+cleared, or when an item is selected.
+
+##### `selectedItem?: Item`
+
+Set the selected item by passing the item you wish to select
+
+##### `initialSelectedItem?: Item`
+
+Pass an item or an array of items that should be selected when a combobox is initialized.
+
+##### `id?: string`
+
+Manually override the id of the combobox trigger
+
+##### `labelId?: string`
+
+Manually override the id of the combobox label
+
+##### `menuId?: string`
+
+Manually override the id of the combobox option list
+
+##### `toggleButtonId?: string`
+
+Manually override the id of the combobox toggle button if one is used
+
+##### `getItemId?: (index: number) => string`
+
+Used for aria attributes and the id prop of the element you use `getInputProps` with.
+
+##### `stateReducer?: (state: UseSelectState<Item>, actionAndChanges: UseSelectStateChangeOptions<Item>) => UseSelectState<Item>`
+
+Called each time a combobox sets its internal state. It allows you to modify the state change that will take place which
+can give you control over how the component interacts with user updates. It gives you the current state and the state
+that will be set, and you return the state that you want to set.
+
+##### `onStateChange?: (changes: Partial<UseSelectState<Item>>) => void`
+
+Called anytime the internal state changes. This can be useful if you're using a combobox as a "controlled" component,
+where you manage some or all of the state and then pass it as props.
+
+##### `environment?: Environment`
+
+A different `window` context from where your JavaScript is running; i.e., an iframe or a shadow-root.
+
+### useComboboxPrimitive Returned
+
+##### `highlightedIndex: number`
+
+The currently highlighted index.
+
+##### `selectedItem: Item`
+
+The value of the currently selected item
+
+##### `isOpen: boolean`
+
+The `isOpen` state
+
+##### `getToggleButtonProps: (options?: UseSelectGetToggleButtonPropsOptions) => any`
+
+Returns the props you should apply to any menu toggle button element you render.
+
+##### `getLabelProps: (options?: UseSelectGetLabelPropsOptions) => any`
+
+Returns the props you should apply to the `label` element that you render.
+
+##### `getMenuProps: (options?: UseSelectGetMenuPropsOptions) => any`
+
+Returns the props you should apply to the `ul` element (or root of your menu) that you render.
+
+##### `getItemProps: (options: UseSelectGetItemPropsOptions<Item>) => any`
+
+Returns the props you should apply to any menu item elements you render.
+
+##### `reset: () => void`
+
+Resets the combobox state to a default.
+
+##### `OpenMenu: () => void`
+
+Opens the menu.
+
+##### `closeMenu: () => void`
+
+Closes the menu.
+
+##### `toggleMenu: () => void`
+
+Toggle the menu open state.
+
+##### `selectItem: (item: Item) => void`
+
+Selects the given item.
+
+##### `setHighlightedIndex: (index: number) => void`
```

Comment:
🤔 I now kind of wish we adopted this props format everywhere...

Reply:
As in, not the table of props?

If so, +1

Reply:
Same. I like this format better. Easier for mobile as well.

Reply:
Yeah.  Let this be our path forward and we will eventually be consistent :D